### ISSUE TITLE
Validation with #[serde(transparent)]

### ISSUE THREAD

**Issue by dimbleby** [GREsau/schemars#107]
I was excited to see the new validation arriving in 0.8.4, as I was hoping that I could get rid of some hand-written parts in my own code.  But I _think_ that currently it doesn't support my use case, so I'm raising this to (i) confirm that's true (ii) check whether you'd consider this to be a bug.

Let's say my API includes a list of strings, where each string should conform to some pattern.  Let's say, they should be numeric strings.

```json
{
   "numeric_strings": ["123", "234"]
}
```

How should I express this in schemars?  I was hoping to write this:

```rust
#[derive(Debug, JsonSchema)]
#[serde(transparent)]
pub struct NumericString(#[schemars(regex(pattern = r"^\d+$"))] String);

#[derive(Debug, JsonSchema)]
pub struct VecOfNumericStrings {
    numeric_strings: Vec<NumericStrings>,
}
```
But the generated schema does not use the pattern at all.

Did I miss a better way?

Thanks!

**Comment by GREsau** [comment]
Currently schemars ignores any attributes on a field of a struct with `#[serde(transparent)]`, which definitely could be considered a bug.

Although in this particular case, I'm not sure you actually need `#[serde(transparent)]`. That attribute is more useful for structs with named fields - for newtype structs like `NumericString`, it rarely has any effect (at least when serializing as JSON).

**Comment by dimbleby** [comment]
You are right, I can simply lose the 'transparent' - thanks!

> Currently schemars ignores any attributes on a field of a struct with #[serde(transparent)]

Not sure that's quite right - the code that I am now able to remove had `#[schemars(schema_with = ...` in such a structure.  (You may remember #67, making this work).

But this is more or less moot, at least for my immediate use case, now that you point out that I can lose the 'transparent' altogether.

**Comment by GREsau** [comment]
That's true, what I should have said was schemars ignores _most_ attributes on a field of a struct with `#[serde(transparent)]`. `with`/`schema_with` are notable exceptions to this

**Comment by baxterjo** [comment]
Found this and it worked for me. Thanks! Might want to add docs to the site for this.

**Comment by GREsau** [comment]
I've just merged a change (https://github.com/GREsau/schemars/pull/415) that should improve this behaviour - when released, validation (and other) attributes will no longer be ignored in transparent structs 🙂
